libcudacxx/include/cuda/std/detail/libcxx/include/__config

```diff
@@ -146,11 +146,24 @@ extern "C++" {
 #    define _LIBCUDACXX_HAS_MSVC_ATOMIC_IMPL
 #  endif
 
-#  define _LIBCUDACXX_ATOMIC_UNSAFE_AUTOMATIC_STORAGE
+// Clang provides 128b atomics as a builtin
+#  if defined(LIBCUDACXX_ENABLE_EXPERIMENTAL_HOST_ATOMICS_128B)
+#    define _CCCL_HOST_128_ATOMICS_ENABLED() 1
+#    define _CCCL_HOST_128_ATOMICS_MAYBE()   0
+// GCC does not provide 128b atomics, but they may be available as a library, this requires opt-in usage.
+// See: https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html "-mcx16" for more
+#  elif _CCCL_COMPILER(CLANG) || _CCCL_COMPILER(GCC)
+#    define _CCCL_HOST_128_ATOMICS_ENABLED() 0
+#    define _CCCL_HOST_128_ATOMICS_MAYBE()   1
+#  else
+#    define _CCCL_HOST_128_ATOMICS_ENABLED() 0
+#    define _CCCL_HOST_128_ATOMICS_MAYBE()   0
+#  endif
+
```

Comment:
We have a `__internal/features.h` I believe that would be the appropriate place for this

Reply:
Moved.